streaming/src/main/scala/org/apache/spark/streaming/receiver/ReceivedBlockHandler.scala

```diff
@@ -199,3 +220,15 @@ private[streaming] object WriteAheadLogBasedBlockHandler {
     new Path(checkpointDir, new Path("receivedData", streamId.toString)).toString
   }
 }
+
+/**
+ * A utility that will wrap the Iterator to get the count
+ */
+private class CountingIterator[T: Manifest](iterator: Iterator[T]) extends Iterator[T] {
+   var count = 0
+   def hasNext(): Boolean = iterator.hasNext
```

Comment:
Add a method 

`def isFullyConsumed = !iterator.hasNext` and use that to detect whether the block was completely consumed (that is, inserted into block manager) or not.